### DISCUSSION TITLE
blocking on the main thread

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#17156] [Q&A]
Hi,
To avoid blocking on the main thread we pass most of the work to a Pthread.
suppose that other module (i.e UI event) has a call to this pthread,  this call should be a sync call (without using the Asyncify because of performance and other issues).
isn't we getting to the same problem - blocking on the main thread ? how can we get over on such case ?

**Comment by sbc100** [comment]
Yes, in the general case you cannot block arbitrarily on the main thread, and that includes synchronously waiting for another thread to do something.

In practice, there are many times that it is safe, for example if you know that the pthread you are waiting on will complete its work in short order.

If only a deadlock if neither thread can make progress.  For example, if the pthread is doing work that requires the main thread to turn its event loop, but the main thread is blocking without returning the event loop.

**Comment by aharondavid** [comment]
Ok, thanks
